pkgs/development/compilers/dart/package-overrides/system-tray/default.nix

```diff
@@ -0,0 +1,18 @@
+{ libayatana-appindicator
+}:
+
+{ ... }:
+
+{ preBuild ? ""
+, ...
+}:
+
+{
+  preBuild = preBuild + ''
+    # $PUB_CACHE/hosted is a symlink to a store path.
+    mv $PUB_CACHE/hosted $PUB_CACHE/hosted_copy
+    cp -HR $PUB_CACHE/hosted_copy $PUB_CACHE/hosted
+    substituteInPlace $PUB_CACHE/hosted/pub.dev/system_tray-*/linux/tray.cc \
```

Comment:
Normally patching the code like this would be a good solution, but modifying the package cache is a bit hacky. Would something like `patchelf --add-rpath` or `--add-needed` not work? The other alternative is adding to LD_LIBRARY_PATH.

This will be obsoleted by #263345 anyway, so I'm not too hung up on which solution we use for now.

Reply:
I was told that such patch is preferred. :) Could you add something like `cargoDepsCopy` to `buildFlutterApplication`?